aten/src/ATen/native/Resize.cpp

```diff
@@ -6,6 +6,20 @@
 
 namespace at { namespace native {
 
+void resize_output(Tensor& output, IntArrayRef shape) {
+  // Tests for resizing of tensors with one more elements
+  if (output.numel() != 0 && !output.sizes().equals(shape)) {
+    TORCH_WARN(
+      "An output with one or more elements was resized, since it had ",
+      "shape ", output.sizes(), ", which does not match the required ",
+      "output shape ", shape, ".",
+      "This behavior is deprecated, and in a future PyTorch release outputs ",
+      "will not be resized unless they have zero elements.");
```

Comment:
I generally like to tell users how to fix warnings in the warning message. Wondering if we can give more specific guidance in the case when `output.numel() == shape.numel()`. In this case, we could tell the user to explicitly view their output so that the warning goes away.